whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/MessageService.java

```diff
@@ -40,43 +41,60 @@ public MessageService(
     public MessageInfo getBouncedMessageInfo(final MimeMessage message) throws MessagingException, IOException {
         return bouncedMessageParser.parse(message);
     }
-    public MessageInfo getUnsubscribedMessageInfo(final MimeMessage message) throws MessagingException, IOException {
+    public MessageInfo getUnsubscribedMessageInfo(final MimeMessage message) throws MessagingException {
         return unsubscribeMessageParser.parse(message);
     }
 
     public void verifyAndSetAsUndeliverable(final MessageInfo message){
-        final String email = outgoingMessageDao.getEmail(message.messageId());
+        final List<String> emails = outgoingMessageDao.getEmails(message.messageId());
 
-        if (isIncorrectMessage(message, email)){
+        if (isIncorrectMessage(message, emails)){
             return;
         }
 
-        LOGGER.debug("Undeliverable message-id {} email {}", message.messageId(), message.emailAddress());
-        emailStatusDao.createEmailStatus(email, EmailStatus.UNDELIVERABLE);
+        LOGGER.debug("Undeliverable message-id {} email {}", message.messageId(), StringUtils.join(message.emailAddresses(), ", "));
+        message.emailAddresses().forEach(email -> emailStatusDao.createEmailStatus(email, EmailStatus.UNDELIVERABLE));
     }
 
     public void verifyAndSetAsUnsubscribed(final MessageInfo message){
-        final String email = outgoingMessageDao.getEmail(message.messageId());
```

Comment:
when you unsubscribe you cannot get multiple emails for a single unsubscribe request?

Reply:
yes, I still want to keep the method signature as it is. 
The difference is just that the message.emailAddresses() is just one. So I prefer to check that this is true and log an error